cmd/argocd/commands/app.go

```diff
@@ -142,7 +143,9 @@ func NewApplicationCreateCommand(clientOpts *argocdclient.ClientOptions) *cobra.
 					c.HelpFunc()(c, args)
 					os.Exit(1)
 				}
-
+				if setFinalizer {
+					app.Finalizers = append(app.Finalizers, "resources-finalizer.argocd.argoproj.io")
```

Comment:
What if that finalizer is already set on the resource? How does the application behave when it has two same finalizers?

Reply:
This is done while creation, there can't be two finalizer, and it works same even if we add two same finalizer